### ISSUE TITLE
add rotation offset and multiplier and  scale  multiplier to mgear_matrixConstraint node to invert values

### ISSUE THREAD

**Issue by miquelcampos** [mgear-dev/mgear_solvers#7]
Hello @JaschaW I will add some extra functionality to the node

rotation offset
negate scaling (or scaler multiplier)

but first I will create the method to use it in apply op module https://github.com/mgear-dev/mgear_core/issues/68

**Comment by JaschaW** [comment]
Hey @miquelcampos just saw the commit for the apply operator of this. Would you like me to tackle the rotation offset and negate scaling?

**Comment by miquelcampos** [comment]
Hello, @JaschaW  do you have time?  :D I was planning  to do it tomorrow or the day after

regarding the negate scale, I was thinking just to add a multiplier so you can negate multiplying by -1

**Comment by JaschaW** [comment]
@miquelcampos I will also be able to look at it tomorrow or the day after. :D I would be happy to do so though? I would need a bit more info about the negate scaling part? Not sure I fully understand what you have in mind for that.

**Comment by miquelcampos** [comment]
Thanks! @JaschaW  I will wait for the update :D 

About the negate scale is just a way to invert some of the values. This is needed for the control_01 full symmetrical behavior because I am using negative scaling in the control to achieve the effect. So that  needs to be "clean" on the joint

For example in this situation
![image](https://user-images.githubusercontent.com/1050212/100714994-ea8e4e00-33f9-11eb-8a8d-b8e519b518b1.png)

Here is the connection to negate the scale. The right and left joint are graphed, Note the difference in the connection. Also NOTE: Also the rotation needs to be inverted 
![image](https://user-images.githubusercontent.com/1050212/100715278-51ac0280-33fa-11eb-8cf9-ad0f7d0d60fe.png)

and all this is because this negation here:
![image](https://user-images.githubusercontent.com/1050212/100715376-743e1b80-33fa-11eb-8521-933374a83e5e.png)

here is the sample scene
https://www.dropbox.com/s/rew7xed0svowdxr/mirror_control_sample.ma?dl=0

So in short we need a multiplier for rotation and scaling. If  just a simple negate is more convenient just let me know

does it make sense?  or do you think that this will be better handle with extra nodes, like before?  Since this is the only situation we need it, maybe is better to keep the node simpler?

**Comment by JaschaW** [comment]
Thanks @miquelcampos this gives me a much better idea of what you are talking about! I actually initially had a different idea of rotation offset.

To me, this looks like we could add a feature to the node that acts similar to a multiply divide node, input the values (X, Y, Z) and have them set to 1 or -1 for both the rotate and scale, that way the node can calculate the correct final output. Are you imagining something similar?

The same for rotate and scale?
![image](https://user-images.githubusercontent.com/31500795/100720410-9fa71300-33c6-11eb-87a6-4f90b60d69d5.png)

One could argue the fact of being explicit with the node, but getting rid of 4 extra nodes, especially the unit conversion nodes seems a big win in my opinion. That is just me in this situation. 
If this is the only situation, I think it might be okay, but if there ever becomes more situations where this could be the better option then I think we should without a doubt.

What are you final thoughts on this? :D I am on the fence for both options!

**Comment by miquelcampos** [comment]
Yes @JaschaW exactly. The multiplier for rotation and scale. 

indeed I was wrong thinking about rotation offset.  both should be multipliers :P

can you add these multipliers?

**Comment by JaschaW** [comment]
@miquelcampos Happy to do so! :)

**Comment by miquelcampos** [comment]
Hello @JaschaW I have changed the node ID and also finally added the rotation offset.
Sorry for the confusion. The rotation offset was needed for this ticket https://github.com/mgear-dev/shifter/issues/73
Do you mind checking my commit? I am not sure If my solution is good.  :P 

Thanks!

**Comment by JaschaW** [comment]
Hey @miquelcampos perfect! Thanks for updating that! Ah so the rotation offset was needed as well, that is what my understanding was originally when we were talking about it ;D Thanks for adding that in man. Checked the commit now and looks good to me! :)

**Comment by miquelcampos** [comment]
@JaschaW thanks for checking it!

And sorry for the confusion. It is my bad because I forgot about that part when we talked :P

**Comment by JaschaW** [comment]
No worries at all @miquelcampos, glad that it is cleared up and we have an awesome new node :)
